### DISCUSSION TITLE
What's the best way to limit memory consumption for a long-running mitmproxy(web) instance?

### DISCUSSION THREAD

**Discussion by dream10201** [mitmproxy/mitmproxy#7288] [Q&A] [answered]
I've configured body_size_limit to '4m', but the process has grown to consume 8.9g of memory after weeks of operation.
![image](https://github.com/user-attachments/assets/ec31ba36-a426-44a7-a3b1-9b249dc75e14)

**Comment by vipertje** [comment, marked as answer]
There is none. Been a long standing issue unfortunately. Other proxies like Charles do have this.
